tests/phpunit/tests/rest-api/rest-blocks-controller.php

```diff
@@ -220,4 +220,39 @@ public function test_content() {
 			$data['content']
 		);
 	}
+
+	/**
+	 * Check that the `wp_pattern_sync_status` postmeta is moved from meta array to top
+	 * level of response.
+	 *
+	 * @ticket 58677
+	 */
+	public function test_wp_patterns_sync_status_post_meta() {
+		register_post_meta(
+			'wp_block',
+			'wp_pattern_sync_status',
+			array(
+				'single'       => true,
+				'type'         => 'string',
+				'show_in_rest' => array(
+					'schema' => array(
+						'type'       => 'string',
+						'properties' => array(
+							'sync_status' => array(
+								'type' => 'string',
+							),
+						),
```

Comment:
```suggestion
						'enum' => array( 'partial', 'unsynced' ),
```

Should this be matching the actual post meta registration?

Reply:
I was just mocking the minimum required to make the test valid, we are not testing the post meta registration, just that the meta field is moved so the only thing that matters in terms of this test is that a field of type string called `wp_pattern_sync_status` is set. 

I don't have a strong opinion on this though, can just copy and paste the full registration if that is seen as better practice.